### DISCUSSION TITLE
Batch run issue analysis on multiple files

### DISCUSSION THREAD

**Discussion by klauskopec** [sn4k3/UVtools#497] [Ideas]
It would be great to be able to run issue analysis as batch and then later being able to simply (and much more quickly) load the saved results per file.

Use case: my montly patreons release pre-supported models and I'd like to verify them using the UVtools issue analyzer. For this, I currently use Lychee's batch feature to slice all of these models (each in their own file) into CTB files. I then need to open one CTB file in UVtools and then have to wait for a few minutes because the issue analysis takes some time on big files.

If we could batch-run the issue analysis, that whole waiting time could be bundled by starting the batch, coming back maybe 1h later and then only having the (relatively short) CTB file load time + issues results loading before I see the results.

**Comment by sn4k3** [comment]
The workflow would be something like batch process files, export results to file, load file in uvtools, load issues from file?
If so this may fit on UVtoolsCmd for batching rather than the UI, the only problem is the issue data structure is not serializable and will require a good effort to make it compatible.

Another option you have right now is going to options and enable auto compute issues on file load. Then you can drag and drop all files into UI and they will all process at same time into a individual UVtools UI instance and compute issues. But it's a big load and keep in mind you will need some heavy resources depending on file count and complexity.

**Reply by klauskopec** [reply]
That's the intended workflow, yes. I'd be super OK with it being in Cmd, but understand that it'd be significant work to implement the serialization.

I'll try the parallel option you described. Let's see how far my 16GB of RAM and 4 yrs old i7 CPU will get me with that ;)
